command/agent/agent_endpoint.go

```diff
@@ -657,3 +658,67 @@ type healthResponseAgent struct {
 	Ok      bool   `json:"ok"`
 	Message string `json:"message,omitempty"`
 }
+
+// AgentHostRequest runs on servers and clients, and captures information about the host system to add
+// to the nomad debug archive.
+func (s *HTTPServer) AgentHostRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	if req.Method != "GET" {
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+
+	var secret string
+	s.parseToken(req, &secret)
+
+	// Check agent read permissions
+	if aclObj, err := s.agent.Client().ResolveToken(secret); err != nil {
```

Comment:
I think we only need to check ACLs on the RPCs, not the servers themselves, This is probably carried over from monitor doing it incorrectly :/

Reply:
Ah. Well I do need to check them because if no node or server is in the query string, I'm returning local host data right here without calling an RPC. I could ... LPC the other endpoint directly which would check though.

Reply:
ah, missed that